editor/components/block-settings-menu/block-transformations.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { noop } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { IconButton } from '@wordpress/components';
+import { getPossibleBlockTransformations, switchToBlockType } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { getBlock } from '../../selectors';
+import { replaceBlocks } from '../../actions';
+
+function BlockTransformations( { blocks, small = false, onTransform, onClick = noop } ) {
+	return <div className="editor-block-settings-menu__block-transformations">
+		{ getPossibleBlockTransformations( blocks ).map( ( { name, title, icon } ) => {
```

Comment:
Given that switchToBlockType receives an array of blocks (and needs to) it may make sense that getPossibleBlockTransformations receive the same array. The places that use one function also use the other so if we use an array of slugs we will probably just add a map to create an array of slugs from the array of blocks and then inside the function, a map in the opposite direction. Also receiving an array blocks leaves the possibility in the future for transformations to depend on the presence of some attribute or another more advanced use case.